src/windows/views/timeline_webview.py

```diff
@@ -2367,13 +2372,12 @@ def Time_Triggered(self, action, clip_ids, speed="1X", playhead_position=0.0):
                     clip.data["reader"]["video_length"] = self.round_to_multiple(
                         float(clip.data["reader"]["video_length"]) / speed_factor, even_multiple)
 
-                if action == MENU_TIME_NONE:
+                if action == MENU_TIME_NONE and "original_data" in clip.data.keys():
                     # Reset original end & duration (if available)
-                    if "original_data" in clip.data.keys():
-                        clip.data["end"] = clip.data["original_data"]["end"]
-                        clip.data["duration"] = clip.data["original_data"]["duration"]
-                        clip.data["reader"]["video_length"] = clip.data["original_data"]["video_length"]
-                        clip.data.pop("original_data")
+                    clip.data["end"] = clip.data["original_data"]["end"]
+                    clip.data["duration"] = clip.data["original_data"]["duration"]
+                    clip.data["reader"]["video_length"] = clip.data["original_data"]["video_length"]
+                    clip.data.pop("original_data")
```

Comment:
This could even be,
```suggestion
                    orig = clip.data.pop("original_data")
                    clip.data.update({
                        "end": orig["end"],
                        "duration": orig["duration"],
                    })
                    clip.data["reader"]["video_length"] = orig["video_length"]
```
(Though I honestly can't fathom how that last line can be correct, regardless. Holy hell.)